Indexing/CodeTree.hpp

```diff
@@ -254,65 +256,70 @@ class CodeTree
     CodeOp* _alternative;
   };
 
+  /**
+   * A search structure that collects alternatives of the same
+   * kind (either function symbols or ground terms) and offers
+   * more efficient searching and insertion over them.
+   */
```

Comment:
Could you say something about *what* the data structure is? IIRC it was something non-obvious.

Reply:
You mean sorted array and a binary search method? I'm not sure what else is there.

Reply:
Ah - OK, that worked in the end? Great stuff, ignore me then. I thought there was some complicated triangular tree thing going on implicitly that I was too stupid to see.